protos/escrow/escrow.proto

```diff
@@ -0,0 +1,101 @@
+syntax = "proto3";
+import "google/protobuf/timestamp.proto";
+
+package btfs;
+
+service EscrowService{
+	rpc SubmitContracts(EscrowContractRequest) returns (EscrowContractResult) {}
+	rpc PayIn(PayinRquest) returns (PayinResult) {}
+	rpc CheckPayment(ContractID) returns (PayinStatus) {}
+	rpc CancelContracts(CancelContractRequest) returns (CancelContractResult) {}
+};
+
+enum TokenType {
+	BTT = 0;
+}
+
+enum Schedule {
+	MONTHLY = 0;
+	QUARTERLY = 1;
+	ANNUALLY = 2;
+}
+
+message ContractID {
+	bytes contract_id = 1;
+}
+
+message EscrowContract {
+	bytes contract_id = 1; // identity of a contract
+	bytes buyer_address = 2;
+	bytes buyer_signature = 3;
+	bytes seller_address = 4;
+	bytes seller_signature = 5; // the seller signature
+	bytes auth_address = 6; // authority who can cancel escrow
+	bytes auth_signature = 7;
+	int64 amount = 8;
+	int64 collateral_amount = 9;
+	int64 withhold_amount = 10; // escrow hold amount till the last layment
+	TokenType token_type = 11; // token type enum
+	Schedule payout_schedule = 12; // schedule enum
+	int32 num_payouts = 13; // numbers of payouts
+	bytes reference = 14; // reference like hash value etc
+}
+
+message EscrowContractRequest {
+	repeated EscrowContract contract = 1;
```

Comment:
Not fully need, but suggest add a field count, to give some verification information.

Reply:
In protobuf standard there's no field count for repeated, like this doc,
https://developers.google.com/protocol-buffers/docs/gotutorial

Reply:
Actually my suggestion is not for protocolbuf, it is related to the verification of the message itself. Thinking the renter's program got some issue for lost one contract while prepare such repeat information, a count field can help both sides understand the message is integrity or not.

Reply:
that's really strange, never saw a protobuf define has count of repeated. we don't need worry too much about integrity.

Reply:
talked to Junlin, nice to have, so not add now.